settings_dialog.js

```diff
@@ -79,14 +94,41 @@ function updateInstalledVersions(){
     });    
 }
 
+function moveInstalledFiles(filePath, destination){
+    
+    for (let i = 0; i < filePath.length; i++) {
+        let file = filePath[i];
+        let fileName = path.basename(filePath[i]);
+
+        movingFileModal.show();
+        let content = document.getElementById("movingFileModalContent");
+        content.innerHTML = "Moving files to: " + destination + " ...";
+        content.append(document.createElement("br"));
+        content.append(document.createTextNode("This may take several minutes, please be patient."))
+        
+        fs.move(file, path.join(destination, fileName))
+        .then(() => {
+            console.log("Successfully moving: " + fileName);
+
+            settings.installPath = destination;
+            onSettingsChanged();
+            updateInstalledVersions();
+            movingFileModal.hide();
```

Comment:
This happens for each successfully moved file? Shouldn't this instead happen after moving all the files?

Reply:
Well, it should. Could I check if all the files have been moved using `if(i == files.length - 1)` inside the loop?

Reply:
Then you would know if the *last* file succeeded or not. And it's not guaranteed that the order is the same. What I'd do is chain the returned promises from the file move. Or just await for each promise to succeed before moving the next file.

Reply:
`fs.move` returns after all files have been moved; not for every file. Promises can only `resolve` or `reject` once.

Reply:
Ah, wait, misread the code itself. Ignore me.

Reply:
This should be fixed as of the latest commit.